### PR TITLE
Expand search box horizontally when text grows long

### LINKED ISSUE #15049: Search box wraps text but it should grow
## Description

The search box wraps long text, which is undesirable because it is very confusing when multi-line search is used.

## Reproduce

Search for:
```
this_is_a_very_very_very_very_long_variable
this_is_a_new_line
```

repro

## Expected behavior

1. The search input grows up until reaching the full width of the container
2. Once full width is reached, it overflows
3. It never wraps

## Context

- Browser and version: Firefox
- JupyterLab version: 4.0.5

### PR DESCRIPTION
**Pull request by sinistersnare**
Hello there! This is my first contribution. Let me know if I should add a test for this (somehow?) or do anything else required. My goal is to learn the codebase a bit and the development workflow here.

## References

Should close #15049. Thanks to #15080 for the work to jump off of!

## Code changes

1. I fixed a typo in docs
2. I added a common hack to naturally grow `<textarea>` elements based on their contents. This utilizes some CSS hackery, but it is [well documented](https://css-tricks.com/auto-growing-inputs-textareas/)

## User-facing changes

Before: [jupyter_search_width_before.webm](https://github.com/jupyterlab/jupyterlab/assets/2153833/ad84acf2-d047-485b-9af6-a14158872a84)

After: [jupyter_search_width_after.webm](https://github.com/jupyterlab/jupyterlab/assets/2153833/558ad790-f7eb-4ed9-bc38-8a74e0ab0a5f)

(the viewer isnt showing up for me and idk how to make it do that)

## Backwards-incompatible changes

No API changes here.

### CODE REVIEW

File: packages/documentsearch/src/searchview.tsx [hunk 1]
```diff
@@ -77,13 +78,20 @@ export interface ISearchKeyBindings {
...
         : props.inputRef?.current;
       if (element) {
-        setRows(element.value.split(/\n/).length);
+        const split = element.value.split(/\n/);
+        // use the longest string out of all lines to compute the width.
+        let longest = split.reduce((a, b) => (a.length > b.length ? a : b), '');
+        if (element.parentNode && element.parentNode instanceof HTMLElement) {
+          element.parentNode.dataset.value = longest;
+        }
+        // element.parentNode?.dataset.value = longest;
```
Comment (sinistersnare) [inline packages/documentsearch/src/searchview.tsx:93]: dead code here, will delete along with any other possible changes that need to be made.

**Review by krassowski — COMMENTED** [review]
Thank you for starting this, it looks like a great solution. It should probably have a brief comment documenting how it works in the CSS code because it relies on a bit of a less-known technique.

> should add a test for this (somehow?)

Yes, that would be appreciated. It could be build upon:

https://github.com/jupyterlab/jupyterlab/blob/0f178d3094f07f4474dc171a9493effbf0b38e90/galata/test/jupyterlab/search.test.ts#L29-L39

Some changes will be needed to limit the growth at certain point, or to ensure that other UI elements do not get squished. In particular currently a very long text will make some buttons unreadable/unclickable:

image

image

**Comment by sinistersnare** [comment]
Hey there, fixed up some of the CSS to work a bit better. Now the buttons next to the search boxes do not shrink, and I added some docs.

Still working on the tests, it seems like the failures are just screenshot failures. Is the protocol to just update them?

In terms of adding a new test I can do that, but would be more comfortable doing that after I get all the tests passing.

**Comment by sinistersnare** [comment]
please update galata snapshots

File: packages/documentsearch/style/base.css [hunk 2]
```diff
@@ -32,6 +32,7 @@
 .jp-DocumentSearch-overlay button {
   background-color: var(--jp-toolbar-background);
   outline: 0;
+  margin-top: 1.5px;
```
Comment (sinistersnare) [inline packages/documentsearch/style/base.css:35]: I tried to address the spacing of the buttons with this. I am not so strong with CSS so if there is a better solution let me know!

**Comment by fcollonval** [comment]
Kicking the CI following the generation of the new snapshots

**Comment by krassowski** [comment]
please update galata snapshots

**Review by krassowski — APPROVED** [review]
Thank you @sinistersnare!

**Comment by krassowski** [comment]
@meeseeksdev please backport to 4.0.x

**Comment by sinistersnare** [comment]
Hey there, I appreciate your help getting this through! Thanks!

**Pull request merged**